src/main/java/io/appform/dropwizard/actors/base/Handler.java

```diff
@@ -85,7 +96,30 @@ public void handleDelivery(String consumerTag, Envelope envelope,
         }
     }
 
-    private MessageMetadata messageProperties(final Envelope envelope) {
-        return new MessageMetadata(envelope.isRedeliver());
+    private Callable<Boolean> getHandleCallable(final Envelope envelope,
+                                                final AMQP.BasicProperties properties,
+                                                final byte[] body) throws IOException {
+        val delayInMs = getDelayInMs(properties);
+        val expired = isExpired(properties);
+        val message = mapper.readValue(body, clazz);
+        return () -> handle(message, messageProperties(envelope, delayInMs), expired);
+    }
+
+    private long getDelayInMs(final AMQP.BasicProperties properties) {
+        return properties.getTimestamp() != null
+                ? Math.max(Instant.now().toEpochMilli() - properties.getTimestamp().getTime(), 0)
+                : -1;
+    }
+
+    private boolean isExpired(final AMQP.BasicProperties properties) {
+        if (properties.getHeaders().containsKey(MESSAGE_EXPIRY_TEXT)) {
```

Comment:
Check for null value in the map

Reply:
Yeah missed it!!!, The existing msgs may not have an header and will fail in that case